### PR TITLE
Adding a standardized pagination implementation to entitlement entities OM-779

### PR DESCRIPTION
**Pull request by GAlexIHU**
## Overview

The aim of this PR is to add a standardized pagination interface (Request Response) to the external and internal APIs. Currently included APIs:
- /api/v1/entitlements
- /api/v1/features
- /api/v1/grants

## Notes for reviewer
- Limit & Offset are kept for BW compatibility (which results in a little mess) but it doesn't break the API. Page has preference over limit&offset. 
- Due to supporting both, page and pagesize have no default values.

### CODE REVIEW

**Review by tothandras — APPROVED** [review]

**Review by tothandras — DISMISSED** [review]

File: api/openapi.yaml [hunk 1]
```diff
@@ -1860,6 +1868,36 @@ components:
         - $ref: "#/components/schemas/FeatureCreateInputs"
         - $ref: "#/components/schemas/SharedMetaFields"
 
+    FeaturePaginatedUnion:
```
Comment (sagikazarmark) [inline api/openapi.yaml:1871]: How about FeatureList or Features or FeatureWuery|ListResults?

File: api/openapi.yaml [hunk 2]
```diff
@@ -3213,6 +3309,26 @@ components:
...
+      name: page
+      in: query
+      required: false
+      description: Page number to return
+      schema:
+        type: integer
+        minimum: 1
+        default: 1
+        example: 1
+    queryPageSize:
```
Comment (sagikazarmark) [inline api/openapi.yaml:3322]: Hos is this different from limit? I think I like that better.

Comment (GAlexIHU) [reply api/openapi.yaml:3322]: You mean semantically? `limit` suggests the resultset will have a certain size, `pageSize` suggests that if i'm on `page` 5 with `pageSize` 20 I would get results 80-99 (i.e. its the "paging increment")

File: api/openapi.yaml (hunk 1 above)
Comment (GAlexIHU) [reply api/openapi.yaml]: i'm fine with `ListFeaturesResponse`, renamed it

**Review by tothandras — APPROVED** [review]
Let's get this merged first

File: api/client/node/clients/entitlement.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { components, operations } from '../schemas/openapi.js'
 import { RequestOptions, BaseClient, OpenMeterConfig } from './client.js'
+import { Paginated } from './pagination.js'
```
Comment (tothandras) [inline api/client/node/clients/entitlement.ts:3]: 🙏🏻 thanks!

**Pull request merged**